doc/examples/transform/plot_tps_deformation.py

```diff
@@ -0,0 +1,142 @@
+r"""
+==========================================
+Interpolate images with thin-plate splines
+==========================================
+
+A conventional technique for interpolating surfaces over a set of data points
+are thin-plate splines (TPS) [1]_ [2]_.
+In an image context, given pairs of source and target control points, TPS can
+be used to transform a space, which, in our case, is a 2D image.
+
+
+.. [1] Wikipedia, Thin plate spline
+       https://en.wikipedia.org/wiki/Thin_plate_spline
+
+.. [2] Bookstein, Fred L. "Principal warps: Thin-plate splines and the
+       decomposition of deformations." IEEE Transactions on pattern analysis and
+       machine intelligence 11.6 (1989): 567–585.
+       :DOI:`10.1109/34.24792`
+       https://user.engineering.uiowa.edu/~aip/papers/bookstein-89.pdf
+
+
+Deform an image
+===============
+
+Image deformation implies displacing the pixels of an image relative to one another.
+In this example, we deform the (2D) image of an astronaut by using thin-plate splines.
+In our image, we define 6 source and target points labeled "1-6": "1-4" are found near
+the image corners, "5" near the left smile corner, and "6" in the right eye.
+At the "1-4" points, there is no displacement.
+Point "5" is displaced upward and point "6" downward.
```

Comment:
I think this is too detailed; simply say that you picked a bunch of source and target points, and describe how you indicate them in the figure.